std/format.d

```diff
@@ -3040,19 +3040,49 @@ if (isInputRange!T)
             static if (is(StringTypeOf!T))
             {
                 auto s = val[0 .. f.precision < $ ? f.precision : $];
+
+                // Avoid expensive call to byGrapheme
+                bool onlyAscii;
+                static if (is(CharTypeOf!E == char))
+                {
+                    onlyAscii = true;
+                    for (size_t i; i < s.length; i++)
+                        if (s[i] > 0x7F)
+                    {
+                        onlyAscii = false;
+                        break;
+                    }
+                }
```

Comment:
I'm not 100% sure this is necessary.  Well, I *did* say yesterday that grapheme clustering in general is expensive, but without profiling the code to be sure that that's actually a problem, I would lean towards not adding more complications to the code than necessary to fix the bug. Besides, the fix proposed here just involves walking the entire string one more time anyway, so it may not actually buy us any extra performance.

Reply:
I would be interested in whether this improves the performance. `std.format` is very heavily used, so imho it's important to avoid introducing a huge performance regression.

Reply:
```d
benchmark!((){auto s = format("%100s","SSßŸé©é©é§SSÃŸß");})(1_000_000)[0].writeln;
```

before : 2 sec 13 ms
after: 3 sec 96 ms

When no width is specified there's no change (after a change just pushed)

Reply:
What about a benchmark to compare the performance of an ASCII string vs. a UTF-8 string without grapheme clusters vs. UTF-8 with grapheme clusters?  Also, compare the performance with and without the `onlyAscii` optimization.

From an initial glance, it looks like the `onlyAscii` optimization might be necessary after all.  2 sec 13 ms to 3 sec 96 ms is a pretty big change.